pyignite/connection/protocol_context.py

```diff
@@ -0,0 +1,99 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple
+
+from pyignite.connection.bitmask_feature import BitmaskFeature
+
+
+class ProtocolContext:
+    """
+    Protocol context. Provides ability to easily check supported supported
+    protocol features.
+    """
+
+    def __init__(self, version: Tuple[int, int, int], features: BitmaskFeature = None):
+        self._version = version
+        self._features = features
+        self._ensure_consistency()
+
+    def __hash__(self):
+        return hash((self._version, self._features))
+
+    def __eq__(self, other):
+        if isinstance(other, ProtocolContext):
+            return self.version == other.version and \
+                   self.features == other.features
+        return NotImplemented
+
+    def _ensure_consistency(self):
+        if not self.is_feature_flags_supported():
+            self._features = None
+
+    @property
+    def version(self):
+        return getattr(self, '_version', None)
+
+    def set_version(self, version: Tuple[int, int, int]):
```

Comment:
Why not use `@version.setter` ?

Reply:
I'm not sure that it's OK when setter has some side-logic. Is it?

Reply:
I suppose that it is main purpose of getters/setters :)

Reply:
Changed it.